pkg/build/builder/source.go

```diff
@@ -109,6 +110,9 @@ func ManageDockerfile(dir string, build *buildapiv1.Build) error {
 			baseDir = filepath.Join(baseDir, build.Spec.Source.ContextDir)
 		}
 		if err := ioutil.WriteFile(filepath.Join(baseDir, "Dockerfile"), []byte(*dockerfileSource), 0660); err != nil {
+			build.Status.Phase = buildapiv1.BuildPhaseFailed
+			build.Status.Reason = buildapiv1.StatusReasonManageDockerfileFailed
+			build.Status.Message = builderutil.StatusMessageManageDockerfileFailed
```

Comment:
you need to make the api call to update the build somewhere in this call stack, e.g.: https://github.com/openshift/builder/blob/adc71a7dbf07918033f3b06b75988a71fef5458f/pkg/build/builder/cmd/builder.go#L239-L240

the ManageDockerfile callstack doesn't appear to currentl do that, so just updating the build object won't propagate the change back to etcd.

Reply:
ah missed that thanks @bparees 

in the update I'm about to push, since `ManageDockerfile` is not associated with the `builderConfig` struct I had to slightly mutate the exact lines of code used, but it was not onerous.